microsoft-edge/devtools-guide-chromium/memory-problems/memory-101.md

```diff
@@ -23,165 +23,146 @@ ms.date: 07/21/2023
+*  Heap objects, referred to as **heap numbers**. Heap numbers are used for storing values that don't fit into the SMI form, such as **doubles**, or when a value needs to be **boxed**, such as setting properties on it.
 
 **Strings** can be stored in either:
 
 *  The **VM heap**.
 
-*  Externally in the **renderer's memory**.  A _wrapper object_ is created and used for accessing external storage where, for example, script sources and other content that is received from the Web is stored, rather than copied onto the VM heap.
+*  Externally in the **renderer's memory**. A _wrapper object_ is created and used for accessing external storage where, for example, script sources and other content that is received from the Web is stored, rather than copied onto the VM heap.
 
-Memory for new JavaScript objects is allocated from a dedicated JavaScript heap (or _VM heap_).  These objects are managed by VM V8's garbage collector, and therefore, these objects stay alive as long as there is at least one strong reference<!-- undefined term --> to them.
+Memory for new JavaScript objects is allocated from a dedicated JavaScript heap (or _VM heap_). These objects are managed by V8 VM's garbage collector, and therefore, these objects stay alive as long as there is at least one strong reference<!-- undefined term --> to them.
 
-**Native objects** - Anything not in the JavaScript heap is called a _native object_.  A native object, in contrast to a heap object, isn't managed by the V8 garbage collector throughout its lifetime, and can only be accessed from JavaScript by using its JavaScript wrapper object.
+**Native objects** - Anything not in the JavaScript heap is called a _native object_. A native object, in contrast to a heap object, isn't managed by the V8 garbage collector throughout its lifetime, and can only be accessed from JavaScript by using its JavaScript wrapper object.
 
-A **cons string** (concatenation string) is an object that consists of pairs of strings that are stored and then joined, and is a result of concatenation.  The joining of the **cons string** contents occurs only as needed.  For example, when a substring of a joined string needs to be constructed.
+A **cons string** (concatenation string) is an object that consists of pairs of strings that are stored and then joined, and is a result of concatenation. The joining of the **cons string** contents occurs only as needed. For example, when a substring of a joined string needs to be constructed.
 
-For example, if you concatenate `a` and `b`, you get a string `(a, b)` which represents the result of concatenation, and is a cons string.  If you later concatenated `d` with that result, you get another cons string: `((a, b, d)`.
+For example, if you concatenate `a` and `b`, you get a string `(a, b)` which represents the result of concatenation, and is a cons string. If you later concatenated `d` with that result, you get another cons string: `((a, b, d)`.
```

Comment:
```suggestion
For example, if you concatenate `a` and `b`, you get a string `(a, b)` which represents the result of concatenation, and is a cons string. If you later concatenated `c` with that result, you get another cons string: `((a, b, c)`.
```
* c - doesn't raise questions, like "d".  Is there some reason "d" was used instead of "c"?